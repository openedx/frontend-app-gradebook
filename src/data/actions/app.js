import { StrictDict } from 'utils';
import { modalFieldKeys, localFilterKeys } from 'data/constants/app';
import { createActionFactory } from './utils';

export const dataKey = 'app';
const createAction = createActionFactory(dataKey);

const closeModal = createAction('closeModal');
/**
 * setCourseId(courseId)
 * loads courseID into local app state
 * @param {string} courseId - course ID from app context
 */
const setCourseId = createAction('setCourseId');

const filterMenu = StrictDict({
  endTransition: createAction('filterMenu/endTransition'),
  startTransition: createAction('filterMenu/startTransition'),
  toggle: createAction('filterMenu/toggle'),
});

/**
  * setModalStateFromTable({ userEntry, subsection })
  * sets modalState
  */
const setModalStateFromTable = createAction('setModalStateFromTable');

/**
 * setSearchValue(searchValue)
 * sets searchValue in local state
 * @param {string} searchValue - search input text
 */
const setSearchValue = createAction('setSearchValue');

const filterReducer = (payload) => (obj, key) => (
  (payload[key] !== undefined) ? { ...obj, [key]: payload[key] } : obj
);

/**
 * setLocalFilter(filterObject)
 * sets a number of localFilterKey fields on localFilters
 * @param {object} filterObject - an object of filter values
 */
const setLocalFilter = createAction('setLocalFilter', (filterObject) => ({
  payload: Object.keys(localFilterKeys).reduce(filterReducer(filterObject), {}),
}));

/**
 * setModalState(modalState)
 * sets a number of modalFieldKey fields on modalState
 * @param {object} modalState - an object of modal state values
 */
const setModalState = createAction('setModalState', (modalState) => ({
  payload: Object.keys(modalFieldKeys).reduce(filterReducer(modalState), {}),
}));

/**
 * setShowImportSuccessToast(shouldShow)
 * Set whether or not to show the Import Grades success toast
 * @param {bool} sholdShow - should show the toast?
 */
const setShowImportSuccessToast = createAction('setShowImportSuccessToast');

/**
 * setView(viewId)
 * sets the UI to display the tab indcated by the passed view id
 * @param {string} viewId - view id as set in app constants
 */
const setView = createAction('setView');

export default StrictDict({
  closeModal,
  filterMenu,
  setCourseId,
  setLocalFilter,
  setModalState,
  setModalStateFromTable,
  setSearchValue,
  setShowImportSuccessToast,
  setView,
});
