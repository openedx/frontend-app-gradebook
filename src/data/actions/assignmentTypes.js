import { StrictDict } from '@src/utils';
import { createActionFactory } from './utils';

export const dataKey = 'assignmentTypes';
const createAction = createActionFactory(dataKey);

const fetching = {
  error: createAction('fetching/error'),
  started: createAction('fetching/started'),
  /**
   * fetching.received(results)
   * @param {object[]} results - assignmentType fetch results
   */
  received: createAction('fetching/received'),
};
/**
 * gotGradesFrozen(areGradesFrozen)
 * @param {bool} are grades frozen?
 */
const gotGradesFrozen = createAction('gotGradesFrozen');

export default StrictDict({
  fetching: StrictDict(fetching),
  gotGradesFrozen,
});
