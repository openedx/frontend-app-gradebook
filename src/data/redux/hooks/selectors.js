import { useSelector } from 'react-redux';

import { StrictDict } from 'utils';
import selectors from 'data/selectors';

const selectorHook = (selector) => () => useSelector(selector);

export const root = StrictDict({
  useEditModalPossibleGrade: selectorHook(selectors.root.editModalPossibleGrade),
  useGetHeadings: selectorHook(selectors.root.getHeadings),
  useGradeExportUrl: selectorHook(selectors.root.gradeExportUrl),
  useInterventionExportUrl: selectorHook(selectors.root.interventionExportUrl),
  useSelectedCohortEntry: selectorHook(selectors.root.selectedCohortEntry),
  useSelectedTrackEntry: selectorHook(selectors.root.selectedTrackEntry),
  useShouldShowSpinner: selectorHook(selectors.root.shouldShowSpinner),
  useShowBulkManagement: selectorHook(selectors.root.showBulkManagement),
  useFilterBadgeConfig: (filterName) => useSelector(
    (state) => selectors.root.filterBadgeConfig(state, filterName),
  ),
});

export const app = StrictDict({
  useActiveView: selectorHook(selectors.app.activeView),
  useAssignmentGradeLimits: selectorHook(selectors.app.assignmentGradeLimits),
  useAreCourseGradeFiltersValid: selectorHook(selectors.app.areCourseGradeFiltersValid),
  useCourseGradeLimits: selectorHook(selectors.app.courseGradeLimits),
  useCourseGradeFilterValidity: selectorHook(selectors.app.courseGradeFilterValidity),
  useCourseId: selectorHook(selectors.app.courseId),
  useModalData: selectorHook(selectors.app.modalData),
  useSearchValue: selectorHook(selectors.app.searchValue),
  useShowImportSuccessToast: selectorHook(selectors.app.showImportSuccessToast),
});

export const assignmentTypes = StrictDict({
  useAllAssignmentTypes: selectorHook(selectors.assignmentTypes.allAssignmentTypes),
  useAreGradesFrozen: selectorHook(selectors.assignmentTypes.areGradesFrozen),
});

export const cohorts = StrictDict({
  useAllCohorts: selectorHook(selectors.cohorts.allCohorts),
  // maybe not needed?
  useCohortsByName: selectorHook(selectors.cohorts.cohortsByName),
});

export const filters = StrictDict({
  useData: selectorHook(selectors.filters.allFilters),
  useIncludeCourseRoleMembers: selectorHook(selectors.filters.includeCourseRoleMembers),
  useSelectableAssignmentLabels: selectorHook(selectors.filters.selectableAssignmentLabels),
  useSelectedAssignmentLabel: selectorHook(selectors.filters.selectedAssignmentLabel),
  useAssignmentType: selectorHook(selectors.filters.assignmentType),
});

export const grades = StrictDict({
  useAllGrades: selectorHook(selectors.grades.allGrades),
  useUserCounts: () => ({
    filteredUsersCount: useSelector(selectors.grades.filteredUsersCount),
    totalUsersCount: useSelector(selectors.grades.totalUsersCount),
  }),
  useGradeData: selectorHook(selectors.grades.gradeData),
  useHasOverrideErrors: selectorHook(selectors.grades.hasOverrideErrors),
  useShowSuccess: selectorHook(selectors.grades.showSuccess),
  useSubsectionGrade: ({ gradeFormat, subsection }) => () => (
    selectors.grades.subsectionGrade[gradeFormat](subsection)
  ),
});

export const roles = StrictDict({
  useCanUserViewGradebook: selectorHook(selectors.roles.canUserViewGradebook),
});

export const tracks = StrictDict({
  useAllTracks: selectorHook(selectors.tracks.allTracks),
  // maybe not needed?
  useTracksByName: selectorHook(selectors.tracks.tracksByName),
});

export default StrictDict({
  app,
  assignmentTypes,
  cohorts,
  filters,
  grades,
  roles,
  tracks,
  root,
});
