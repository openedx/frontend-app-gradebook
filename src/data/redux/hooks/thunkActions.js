import { StrictDict } from 'utils';
import thunkActions from 'data/thunkActions';
import { actionHook } from './utils';

const app = StrictDict({
  filterMenu: {
    useCloseMenu: actionHook(thunkActions.app.filterMenu.close),
    useHandleTransitionEnd: actionHook(thunkActions.app.filterMenu.handleTransitionEnd),
    useToggleMenu: actionHook(thunkActions.app.filterMenu.toggle),
  },
  useSetModalStateFromTable: actionHook(thunkActions.app.setModalStateFromTable),
});

const grades = StrictDict({
  useFetchGradesIfAssignmentGradeFiltersSet: actionHook(
    thunkActions.grades.fetchGradesIfAssignmentGradeFiltersSet,
  ),
  useFetchPrevNextGrades: actionHook(thunkActions.grades.fetchPrevNextGrades),
  useFetchGrades: actionHook(thunkActions.grades.fetchGrades),
  useSubmitImportGradesButtonData: actionHook(thunkActions.grades.submitImportGradesButtonData),
  useUpdateGrades: actionHook(thunkActions.grades.updateGrades),
});

export default StrictDict({
  app,
  grades,
});
