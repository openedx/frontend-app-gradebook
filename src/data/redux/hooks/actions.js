import { StrictDict } from 'utils';
import actions from 'data/actions';
import { actionHook } from './utils';

const app = StrictDict({
  useSetLocalFilter: actionHook(actions.app.setLocalFilter),
  useSetSearchValue: actionHook(actions.app.setSearchValue),
  useSetShowImportSuccessToast: actionHook(actions.app.setShowImportSuccessToast),
  useSetView: actionHook(actions.app.setView),
  useCloseModal: actionHook(actions.app.closeModal),
  useSetModalState: actionHook(actions.app.setModalState),
});

const filters = StrictDict({
  useUpdateAssignment: actionHook(actions.filters.update.assignment),
  useUpdateAssignmentLimits: actionHook(actions.filters.update.assignmentLimits),
  useUpdateAssignmentType: actionHook(actions.filters.update.assignmentType),
  useUpdateCohort: actionHook(actions.filters.update.cohort),
  useUpdateCourseGradeLimits: actionHook(actions.filters.update.courseGradeLimits),
  useUpdateIncludeCourseRoleMembers: actionHook(actions.filters.update.includeCourseRoleMembers),
  useUpdateTrack: actionHook(actions.filters.update.track),
  useResetFilters: actionHook(actions.filters.reset),
});

const grades = StrictDict({
  useDoneViewingAssignment: actionHook(actions.grades.doneViewingAssignment),
  useDownloadBulkGradesReport: actionHook(actions.grades.downloadReport.bulkGrades),
  useDownloadInterventionReport: actionHook(actions.grades.downloadReport.intervention),
  useToggleGradeFormat: actionHook(actions.grades.toggleGradeFormat),
  useCloseBanner: actionHook(actions.grades.banner.close),
});

export default StrictDict({
  app,
  filters,
  grades,
});
