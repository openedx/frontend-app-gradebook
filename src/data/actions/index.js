import { StrictDict } from '../../utils';

import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import config from './config';
import filters from './filters';
import grades from './grades';
import roles from './roles';
import tracks from './tracks';

export default StrictDict({
  app,
  assignmentTypes,
  cohorts,
  config,
  filters,
  grades,
  roles,
  tracks,
});
