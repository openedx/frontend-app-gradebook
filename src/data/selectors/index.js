import { StrictDict } from 'utils';

import lms from 'data/services/lms';
import * as filterConstants from 'data/constants/filters';

import * as module from '.';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import filters from './filters';
import grades, { minGrade, maxGrade } from './grades';
import roles from './roles';
import tracks from './tracks';

const {
  filterConfig,
  filters: filterNames,
} = filterConstants;

/**
 * editModalPossibleGrade(state)
 * Returns the "possible" grade as shown in the edit modal.
 * @param {object} state - redux state;
 * @return {string} - possibleGrade to show on edit modal
 */
export const editModalPossibleGrade = (state) => (
  app.modalState.adjustedGradePossible(state) || grades.gradeOriginalPossibleGraded(state)
);

/**
 * filterBadgeConfig(state, name)
 * Takes a filter name and returns the appropriate badge config, with value and isDefault.
 * Determines if it should return a range or single-value config based on the presence of
 * a filterOrder prop in the filter config associated with the passed name.
 * @param {object} state - redux state
 * @param {string} name - api filter name
 */
export const filterBadgeConfig = (state, name) => {
  const filterValue = module.filterBadgeValues[name](state);
  const { filterOrder, ...config } = filterConfig[name];
  const isRange = !!filterOrder;
  const value = isRange ? `${filterValue[0]} - ${filterValue[1]}` : filterValue;
  const isDefault = (isRange
    ? (
      filters.isDefault(filterOrder[0], filterValue[0])
      && filters.isDefault(filterOrder[1], filterValue[1])
    )
    : filters.isDefault(name, filterValue)
  );
  return { ...config, value, isDefault };
};

/**
 * filterBadgeValues methods
 * For each filter type with an associated badge, provides a selector that returns the
 * content of that badge
 */
export const filterBadgeValues = StrictDict({
  [filterNames.assignment]: (state) => (
    filters.selectedAssignmentLabel(state) || ''
  ),
  [filterNames.assignmentType]: filters.assignmentType,
  [filterNames.includeCourseRoleMembers]: filters.includeCourseRoleMembers,
  [filterNames.cohort]: (state) => {
    const entry = module.selectedCohortEntry(state);
    return entry ? entry.name : '';
  },
  [filterNames.track]: (state) => {
    const entry = module.selectedTrackEntry(state);
    return entry ? entry.name : '';
  },
  [filterNames.assignmentGrade]: (state) => ([
    filters.assignmentGradeMin(state),
    filters.assignmentGradeMax(state),
  ]),
  [filterNames.courseGrade]: (state) => ([
    filters.courseGradeMin(state),
    filters.courseGradeMax(state),
  ]),
});

/**
 * formattedGradeLimits(state)
 * Returns an object of local grade limits, formatted for fetching.
 * This means only setting any of them if the assignment is set, and only setting an
 * individual min/max if they are not equal to the default min/max.
 * @param {object} state - redux state
 * @return {object} - { assignmentGradeMax, assignmentGradeMin, courseGradeMax, courseGradeMin }
 */
export const formattedGradeLimits = (state) => {
  const { assignmentGradeMax, assignmentGradeMin } = app.assignmentGradeLimits(state);
  const { courseGradeMax, courseGradeMin } = app.courseGradeLimits(state);
  const hasAssignment = filters.selectedAssignmentId(state) !== undefined;

  return {
    assignmentGradeMax: (assignmentGradeMax === maxGrade || !hasAssignment) ? null : assignmentGradeMax,
    assignmentGradeMin: (assignmentGradeMin === minGrade || !hasAssignment) ? null : assignmentGradeMin,
    courseGradeMax: courseGradeMax === maxGrade ? null : courseGradeMax,
    courseGradeMin: courseGradeMin === minGrade ? null : courseGradeMin,
  };
};

/**
 * getHeadings(state)
 * Returns the table headings given the current assignmentType and assignmentLabel filters.
 * @param {object} state - redux state
 * @return {string[]} - array of table headings
 */
export const getHeadings = (state) => grades.headingMapper(
  filters.assignmentType(state) || 'All',
  filters.selectedAssignmentLabel(state) || 'All',
)(grades.getExampleSectionBreakdown(state));

/**
 * gradeExportUrl(state, options)
 * Returns the output of getGradeCsvUrl, applying the current includeCourseRoleMembers
 * filter.
 * @param {object} state - redux state
 * @return {string} - generated grade export url
 */
export const gradeExportUrl = (state) => (
  lms.urls.gradeCsvUrl(module.lmsApiServiceArgs(state))
);

/**
 * interventionExportUrl(state, options)
 * Returns the output of getInterventionExportUrl.
 * @param {object} state - redux state
 * @return {string} - generated intervention export url
 */
export const interventionExportUrl = (state) => (
  lms.urls.interventionExportCsvUrl(module.lmsApiServiceArgs(state))
);

/**
 * lmsApiServiceArgs(state)
 * Returns common lms api service request args.
 * @param {object} state - redux state
 * @return {object} lms api query params object
 */
export const lmsApiServiceArgs = (state) => ({
  cohort: cohorts.getCohortNameById(state, filters.cohort(state)),
  track: filters.track(state),
  assignment: filters.selectedAssignmentId(state),
  assignmentType: filters.assignmentType(state),
  assignmentGradeMin: grades.formatMinAssignmentGrade(
    filters.assignmentGradeMin(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  assignmentGradeMax: grades.formatMaxAssignmentGrade(
    filters.assignmentGradeMax(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  courseGradeMin: grades.formatMinCourseGrade(filters.courseGradeMin(state)),
  courseGradeMax: grades.formatMaxCourseGrade(filters.courseGradeMax(state)),
  excludedCourseRoles: filters.excludedCourseRoles(state),
});

/**
 * localFilters(state)
 * returns local filter data for fetchGrades call
 * @param {object} state - redux state
 * @return {object} - fetch arguments signifying current local filter state
 */
export const localFilters = (state) => {
  const id = filters.selectedAssignmentId(state);
  const searchText = app.searchValue(state);
  return {
    assignment: id,
    includeCourseRoleMembers: filters.includeCourseRoleMembers(state),
    ...module.formattedGradeLimits(state),
    ...(searchText !== '' && { searchText }),
  };
};

/**
 * selectedCohortEntry(state)
 * Returns the full entry data for the selected cohort
 * @param {object} state - redux state
 * @return {object} - selected cohort entry object
 */
export const selectedCohortEntry = (state) => (
  cohorts.allCohorts(state).find(
    ({ id }) => id === parseInt(filters.cohort(state), 10),
  )
);

/**
 * selectedTrackEntry(state)
 * Returns the full entry data for the selected track
 * @param {object} state - redux state
 * @return {object} - selected track entry object
 */
export const selectedTrackEntry = (state) => (
  tracks.allTracks(state).find(
    ({ slug }) => slug === filters.track(state),
  )
);

/**
 * shouldShowSpinner(state)
 * Returns true iff the user can view the gradebook and grades.showSpinner is true.
 * @param {object} state - redux state
 * @return {bool} - should show spinner?
 */
export const shouldShowSpinner = (state) => (
  roles.canUserViewGradebook(state)
  && grades.showSpinner(state)
);

/**
 * showBulkManagement(state, options)
 * Returns true iff the course has bulk management enabled
 * @param {object} state - redux state
 * @return {bool} - should show bulk management controls?
 */
export const showBulkManagement = (state) => (state.config.bulkManagementAvailable);

export default StrictDict({
  root: StrictDict({
    editModalPossibleGrade,
    filterBadgeConfig,
    getHeadings,
    gradeExportUrl,
    interventionExportUrl,
    localFilters,
    selectedCohortEntry,
    selectedTrackEntry,
    shouldShowSpinner,
    showBulkManagement,
  }),
  app,
  assignmentTypes,
  cohorts,
  filters,
  grades,
  roles,
  tracks,
});
