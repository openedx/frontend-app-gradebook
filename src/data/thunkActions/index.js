import { StrictDict } from '../../utils';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import grades from './grades';
import roles from './roles';
import tracks from './tracks';

export default StrictDict({
  app,
  assignmentTypes,
  cohorts,
  grades,
  roles,
  tracks,
});
