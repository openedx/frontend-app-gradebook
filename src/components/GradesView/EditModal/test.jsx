import React from 'react';
import { shallow } from 'enzyme';

import actions from 'data/actions';
import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';

import {
  EditModal,
  mapDispatchToProps,
  mapStateToProps,
} from '.';

jest.mock('./OverrideTable', () => 'OverrideTable');
jest.mock('./ModalHeaders', () => 'ModalHeaders');
jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
  Modal: () => 'Modal',
  Alert: () => 'Alert',
}));
jest.mock('data/actions', () => ({
  __esModule: true,
  default: {
    app: { closeModal: jest.fn() },
    grades: { doneViewingAssignment: jest.fn() },
  },
}));
jest.mock('data/thunkActions', () => ({
  __esModule: true,
  default: {
    grades: { updateGrades: jest.fn() },
  },
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    app: {
      modalState: {
        open: jest.fn(state => ({ isModalOpen: state })),
      },
    },
    grades: {
      gradeOverrideHistoryError: jest.fn(state => ({ overrideHistoryError: state })),
    },
  },
}));
describe('EditMoal', () => {
  let props;
  beforeEach(() => {
    props = {
      gradeOverrideHistoryError: 'Weve been trying to contact you regarding...',
      open: true,
      closeModal: jest.fn(),
      doneViewingAssignment: jest.fn(),
      updateGrades: jest.fn(),
    };
  });

  describe('Component', () => {
    describe('behavior', () => {
      let el;
      beforeEach(() => {
        el = shallow(<EditModal {...props} />);
      });
      describe('closeAssignmentModal', () => {
        it('calls props.doneViewingAssignment and props.closeModal', () => {
          el.instance().closeAssignmentModal();
          expect(props.doneViewingAssignment).toHaveBeenCalledWith();
          expect(props.closeModal).toHaveBeenCalledWith();
        });
      });
      describe('handleAdjustedGradeClick', () => {
        it('calls props.updateGardes and this.closeAssignmentModal', () => {
          el.instance().closeAssignmentModal = jest.fn();
          el.instance().handleAdjustedGradeClick();
          expect(props.updateGrades).toHaveBeenCalledWith();
          expect(el.instance().closeAssignmentModal).toHaveBeenCalledWith();
        });
      });
    });
    describe('snapshots', () => {
      let el;
      beforeEach(() => {
        el = shallow(<EditModal {...props} />);
        el.instance().closeAssignmentModal = jest.fn().mockName('this.closeAssignmentModal');
        el.instance().handleAdjustedGradeClick = jest.fn().mockName(
          'this.handleAdjustedGradeClick',
        );
      });
      describe('gradeOverrideHistoryError is and empty and open is true', () => {
        test('modal open and StatusAlert showing', () => {
          expect(el.instance().render()).toMatchSnapshot();
        });
      });
      describe('gradeOverrideHistoryError is empty and open is false', () => {
        test('modal closed and StatusAlert closed', () => {
          el.setProps({ open: false, gradeOverrideHistoryError: '' });
          expect(el.instance().render()).toMatchSnapshot();
        });
      });
    });
  });

  describe('mapStateToProps', () => {
    const testState = { martha: 'why did you say that name?!' };
    let mapped;
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('gradeOverrideHistoryError from grades.gradeOverrideHistoryError', () => {
      expect(
        mapped.gradeOverrideHistoryError,
      ).toEqual(selectors.grades.gradeOverrideHistoryError(testState));
    });
    test('open from app.modalState.open', () => {
      expect(mapped.open).toEqual(selectors.app.modalState.open(testState));
    });
  });
  describe('mapDispatchToProps', () => {
    test('closeModal from actions.app.closeModal', () => {
      expect(mapDispatchToProps.closeModal).toEqual(actions.app.closeModal);
    });
    test('doneViewingAssignemtn from actions.grades.doneViewingAssignment', () => {
      expect(
        mapDispatchToProps.doneViewingAssignment,
      ).toEqual(actions.grades.doneViewingAssignment);
    });
    test('updateGrades from thunkActions.grades.updateGrades', () => {
      expect(mapDispatchToProps.updateGrades).toEqual(thunkActions.grades.updateGrades);
    });
  });
});
