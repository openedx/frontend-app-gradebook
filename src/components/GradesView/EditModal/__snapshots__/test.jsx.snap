// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMoal Component snapshots gradeOverrideHistoryError is and empty and open is true modal open and StatusAlert showing 1`] = `
<Modal
  body={
    <div>
      <ModalHeaders />
      <Alert
        dismissible={false}
        show={true}
        variant="danger"
      >
        Weve been trying to contact you regarding...
      </Alert>
      <OverrideTable />
      <div>
        <FormattedMessage
          defaultMessage="Showing most recent actions (max 5). To see more, please contact support"
          description="Edit Modal visibility hint message"
          id="gradebook.GradesView.EditModal.contactSupport"
        />
      </div>
      <div>
        <FormattedMessage
          defaultMessage="Note: Once you save, your changes will be visible to students."
          description="Edit Modal saved changes effect hint"
          id="gradebook.GradesView.EditModal.saveVisibility"
        />
      </div>
    </div>
  }
  buttons={
    Array [
      <Button
        onClick={[MockFunction this.handleAdjustedGradeClick]}
        variant="primary"
      >
        <FormattedMessage
          defaultMessage="Save Grades"
          description="Edit Modal Save button label"
          id="gradebook.GradesView.EditModal.saveGrade"
        />
      </Button>,
    ]
  }
  closeText={
    <FormattedMessage
      defaultMessage="Cancel"
      description="Edit Modal close button text"
      id="gradebook.GradesView.EditModal.closeText"
    />
  }
  onClose={[MockFunction this.closeAssignmentModal]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="Edit Grades"
      description="Edit Modal title"
      id="gradebook.GradesView.EditModal.title"
    />
  }
/>
`;

exports[`EditMoal Component snapshots gradeOverrideHistoryError is empty and open is false modal closed and StatusAlert closed 1`] = `
<Modal
  body={
    <div>
      <ModalHeaders />
      <Alert
        dismissible={false}
        show={false}
        variant="danger"
      >
        
      </Alert>
      <OverrideTable />
      <div>
        <FormattedMessage
          defaultMessage="Showing most recent actions (max 5). To see more, please contact support"
          description="Edit Modal visibility hint message"
          id="gradebook.GradesView.EditModal.contactSupport"
        />
      </div>
      <div>
        <FormattedMessage
          defaultMessage="Note: Once you save, your changes will be visible to students."
          description="Edit Modal saved changes effect hint"
          id="gradebook.GradesView.EditModal.saveVisibility"
        />
      </div>
    </div>
  }
  buttons={
    Array [
      <Button
        onClick={[MockFunction this.handleAdjustedGradeClick]}
        variant="primary"
      >
        <FormattedMessage
          defaultMessage="Save Grades"
          description="Edit Modal Save button label"
          id="gradebook.GradesView.EditModal.saveGrade"
        />
      </Button>,
    ]
  }
  closeText={
    <FormattedMessage
      defaultMessage="Cancel"
      description="Edit Modal close button text"
      id="gradebook.GradesView.EditModal.closeText"
    />
  }
  onClose={[MockFunction this.closeAssignmentModal]}
  open={false}
  title={
    <FormattedMessage
      defaultMessage="Edit Grades"
      description="Edit Modal title"
      id="gradebook.GradesView.EditModal.title"
    />
  }
/>
`;
